src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/PropertyDescriptor.cs

```diff
@@ -165,10 +167,11 @@ public virtual void AddValueChanged(object component, EventHandler handler)
             ArgumentNullException.ThrowIfNull(component);
             ArgumentNullException.ThrowIfNull(handler);
 
-            _valueChangedHandlers ??= new Dictionary<object, EventHandler?>();
-
-            EventHandler? h = _valueChangedHandlers.GetValueOrDefault(component, defaultValue: null);
-            _valueChangedHandlers[component] = (EventHandler?)Delegate.Combine(h, handler);
+            lock (s_internalSyncObject)
```

Comment:
Perhaps we can use `LazyInitializer. EnsureInitialized` here? Then the sync lock doesn't have to be allocated until needed.

Reply:
Sure. Thanks